main/lsp/lsp.h

```diff
@@ -1,18 +1,11 @@
 #ifndef RUBY_TYPER_LSPLOOP_H
 #define RUBY_TYPER_LSPLOOP_H
 
-#include "ast/ast.h"
-#include "common/kvstore/KeyValueStore.h"
-#include "core/ErrorQueue.h"
-#include "core/NameHash.h"
```

Comment:
I noticed that `lsp.h` no longer needs to include many of these headers as I moved the relevant references to other classes, like `LSPTypechecker`.